src/xrootiface.py

```diff
@@ -86,119 +103,129 @@ def init(inconfig, inlog):
   else:
     homepath = ''
 
-def stat(endpoint, filename, ruid, rgid):
-  '''Stat a file via xroot on behalf of the given uid,gid, and returns (size, mtime). Uses the default xroot API.'''
-  filename = _getfilename(filename)
+
+def stat(endpoint, filepath, userid):
+  '''Stat a file via xroot on behalf of the given userid, and returns (size, mtime). Uses the default xroot API.'''
+  filepath = _getfilepath(filepath)
   tstart = time.clock()
-  rc, statInfo = _getxrdfor(endpoint).stat(filename + _eosargs(ruid, rgid))
+  rc, statInfo = _getxrdfor(endpoint).stat(filepath + _eosargs(userid))
   tend = time.clock()
-  log.info('msg="Invoked stat" filename="%s" elapsedTimems="%.1f"' % (filename, (tend-tstart)*1000))
+  log.info('msg="Invoked stat" filepath="%s" elapsedTimems="%.1f"' % (filepath, (tend-tstart)*1000))
   if statInfo is None:
     raise IOError(rc.message.strip('\n'))
   return {'size': statInfo.size, 'mtime': statInfo.modtime}
 
-def statx(endpoint, filename, ruid, rgid):
-  '''Get extended stat info (inode, ouid, ogid, size, mtime) via an xroot opaque query on behalf of the given uid,gid'''
-  filename = _getfilename(filename)
+
+def statx(endpoint, filepath, userid):
+  '''Get extended stat info (inode, filepath, userid, size, mtime) via an xroot opaque query on behalf of the given userid'''
   tstart = time.clock()
-  rc, rawinfo = _getxrdfor(endpoint).query(QueryCode.OPAQUEFILE, filename + _eosargs(ruid, rgid) + '&mgm.pcmd=stat')
+  rc, info = _getxrdfor(endpoint).query(QueryCode.OPAQUEFILE, _getfilepath(filepath) + _eosargs(userid) + '&mgm.pcmd=stat')
```

Comment:
What does rc stand for?

Reply:
Short for return code (quite common within C/C++ code)

Reply:
return code as what is actually returned? Is this best practice to use in python as well?

Reply:
not sure I understand - this `query()` method returns two values, a return status code to say whether the call was successful or not and the stat information. The xrootd python bindings are at https://github.com/xrootd/xrootd-python if you want to have a deeper look. Again I wouldn't bother in the context of this PR.

Reply:
I'm just curious what's inside the "return code", if it's only a status I think it should be called status. Since we're going through a large part of the code I think it's a good opportunity to really be a boy scout :)

Reply:
In my mind, return code and status are just synonyms, in particular when the type is `int` (an enum of constants). And it has been standard practice to name statuses returned by functions just `rc` in lots of places: just out of curiosity, even though Reva typically uses `res` to store the response of function calls, there's at least one place that reads `rc, err = ...`.
For this particular case, if you want to learn more about the actual values returned by the xroot functions they are documented at the link above - and for `query()`, IIRC the first returned value is an enum, so `rc` is actually appropriate (as it would be `status`).

Reply:
OK, Im' just unused to this syntax.